app.js

```diff
@@ -32,69 +35,94 @@ function outputStatus() {
   });
 }
 
-function startRequest() {
-  currentRequests++;
-  var objectToCopy = objectsToCopy.pop();
-
-  s3.copyObject({
-    Bucket: argv.target,
-    CopySource: encodeURIComponent(argv.source) + '/' + objectToCopy.Key,
-    Key: objectToCopy.Key,
-    ACL: argv.acl,
-    MetadataDirective: 'COPY',
-    StorageClass: argv.storageClass || objectToCopy.StorageClass
+function checkKeyMatch(object) {
+  if (!argv.keymatch) {
+    return object;
+  } else {
+    if (object.Key.match(argv.keyMatch)) {
+      return object;
+    } else {
+      return false;
+    }
+  }
+}
+
+function getNextListChunk() {
+  // Read list of objects to copy from source bucket
+  listInProgress = true;
+  
+  s3.listObjects({
+    Bucket: argv.source,
+    Marker: nextMarker
```

Comment:
the `nextMarker` looks to be first set after this is called, is it ok to pass an undefined value to the `s3.listObjects`?